### PR TITLE
feat: comprehensive test suite + spawner CLI refactor (#43)

### PR DESCRIPTION
**Pull request by travisgalloway**
## Summary
- Adds 78 new tests across all three packages (`@caw/core`, `@caw/mcp-server`, `@caw/tui`), expanding total coverage from 950 to 1028 tests across 78 files
- **Core (578 tests, +14):** WAL mode & busy timeout pragmas, session promotion edge cases, agent stale threshold variations, lock guard `isLockedByOther` coverage, and 7 cross-service integration tests covering full workflow lifecycle, replanning, multi-agent coordination, locking, template round-trip, orchestration parallelism, and context recovery
- **MCP Server (198 tests, +16):** tool naming convention validation, config parsing (`parsePort`, `parseTransport`, `parseDbMode`), broadcast with empty filter, auto-mark-read on `message_get`, unread count by priority breakdown, `toolResult` formatting for null/primitives/arrays, and lock-guard suggestion assertions
- **TUI (252 tests, +48):** 14 new component test files (HelpView, WorkflowHeader, SetupGuide, HintBar, ScrollArea, StatsSummary, SystemInfo, WorkflowListScreen, WorkflowDetailScreen, TaskDetailScreen, TaskTree, AgentDetail, MessageDetailScreen, MessageInbox), plus expanded store cycling tests, CommandPrompt feedback clearing, SelectableTable column/truncation tests, and parseCommand `/resume` handling
- **Spawner (43 tests, 7 files):** Agent session lifecycle, pool management, MCP config generation, workflow spawner start/suspend/resume/shutdown, registry, and integration tests (single task, sequential deps, parallel fan-in, suspend/resume)
- **SSE→HTTP transport rename:** Renamed transport type from `sse` to `http` across config schema, server, daemon, docs, and all tests
- **Bug fixes:** Fixed `claude -p` agent spawn hanging by using `stdio: ['ignore', 'pipe', 'pipe']` (piped stdin caused indefinite hang); added `Bun.serve` `idleTimeout: 255` to prevent MCP request timeouts
- **Docs:** Updated all documentation to reflect final state — added `@caw/spawner` to monorepo structure, updated MCP tool count from 43→56 across 12 categories, documented lock/unlock/replanning/execution tools, updated CLI reference with `init`/`setup`/`run` subcommands, removed stale `examples/spawner-test/` directory
- **E2E verified:** Full spawner workflow executed at `/tmp/caw-test` — 4 tasks completed via `claude -p` agents connecting over HTTP MCP transport

## Test plan
- [x] `bun run build` — full monorepo typecheck passes
- [x] `bun run test` — all 1071 tests pass (578 core + 198 mcp-server + 252 tui + 43 spawner)
- [x] `bun run lint` — no lint errors
- [x] E2E: `caw run <workflow_id>` spawns agents that connect via `type: 'http'` MCP config and complete all tasks

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/mcp-server/src/tools/messaging.test.ts [hunk 1]
```diff
@@ -277,6 +322,33 @@ describe('messaging tools', () => {
...
+        body: 'Urgent msg',
+        priority: 'high',
+      });
+
+      const result = call('message_count_unread', { agent_id: recipient });
+      expect(result.isError).toBeUndefined();
+      const data = parseContent(result) as { count: number; by_priority: Record<string, number> };
+      expect(data.count).toBe(2);
+      expect(data.by_priority).toBeDefined();
+    });
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply packages/mcp-server/src/tools/messaging.test.ts]: ✅ Fixed in 62cb88b — now asserts `{ normal: 1, high: 1 }` instead of just `toBeDefined()`

File: packages/core/src/db/connection.test.ts [hunk 2]
```diff
@@ -18,6 +18,34 @@ describe('createConnection', () => {
...
+    try {
+      const db = createConnection(tmpPath);
+      const result = db.prepare('PRAGMA journal_mode').get() as { journal_mode: string };
+      expect(result.journal_mode).toBe('wal');
+      db.close();
+    } finally {
+      const { unlinkSync } = require('node:fs');
+      try {
+        unlinkSync(tmpPath);
+      } catch {}
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply packages/core/src/db/connection.test.ts]: ✅ Fixed in 62cb88b — replaced `require` with top-level `import { rmSync }` and `rmSync(..., { force: true })`

File: apps/tui/src/components/WorkflowListScreen.test.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { WorkflowListScreen } from './WorkflowListScreen';
+
+describe('WorkflowListScreen', () => {
+  test('exports WorkflowListScreen as a function component', () => {
+    expect(typeof WorkflowListScreen).toBe('function');
+  });
+
+  test('has the expected function name', () => {
+    expect(WorkflowListScreen.name).toBe('WorkflowListScreen');
+  });
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply apps/tui/src/components/WorkflowListScreen.test.tsx]: 💬 Reviewed — These are intentionally lightweight smoke tests. Full render tests for these components require mocking the Zustand store, database services, and navigation stack, which is out of scope for this PR. The `.name` check ensures the export identity is stable.

File: packages/mcp-server/src/tools/messaging.test.ts [hunk 4]
```diff
@@ -258,6 +258,51 @@ describe('messaging tools', () => {
...
+  describe('message_broadcast edge cases', () => {
+    it('returns empty result when no agents match filter', () => {
+      const sender = registerAgent('Broadcaster');
+      // No other agents exist — broadcast goes to nobody
+
+      const result = call('message_broadcast', {
+        sender_id: sender,
+        message_type: 'broadcast',
+        body: 'Echo',
+        filter: { role: 'coordinator' },
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply packages/mcp-server/src/tools/messaging.test.ts]: ✅ Fixed in 62cb88b — renamed `filter` to `recipient_filter` and added a non-matching agent to verify filtering actually works
